build-tools/packages/build-cli/src/handlers/checkFunctions.ts

```diff
@@ -477,6 +478,91 @@ export const checkAssertTagging: StateHandlerFunction = async (
 	return true;
 };
 
+/**
+ * Checks that release notes have been generated.
+ *
+ * @param state - The current state machine state.
+ * @param machine - The state machine.
+ * @param testMode - Set to true to run function in test mode.
+ * @param log - A logger that the function can use for logging.
+ * @param data - An object with handler-specific contextual data.
+ * @returns True if the state was handled; false otherwise.
+ */
+export const checkReleaseNotes: StateHandlerFunction = async (
+	state: MachineState,
+	machine: Machine<unknown>,
+	testMode: boolean,
+	log: CommandLogger,
+	data: FluidReleaseStateHandlerData,
+): Promise<boolean> => {
+	if (testMode) return true;
+
+	const { bumpType, releaseGroup, releaseVersion } = data;
+
+	// This check should only be run for client/server minor/major releases. Patch releases should always pass this
+	// check, as will release groups other than client/server.
+	if (["client", "server"].includes(releaseGroup) && bumpType !== "patch") {
+		// Check if the release notes file exists
+		const filename = `RELEASE_NOTES/${releaseVersion}.md`;
+
+		if (!existsSync(filename)) {
+			log.logHr();
+			log.errorLog(
+				`Release notes for ${releaseGroup} version ${releaseVersion} are not found.`,
+			);
+			BaseStateHandler.signalFailure(machine, state);
+			return false;
+		}
+	}
+
+	BaseStateHandler.signalSuccess(machine, state);
+	return true;
+};
+
+/**
+ * Checks that changelogs have been generated.
+ *
+ * @param state - The current state machine state.
+ * @param machine - The state machine.
+ * @param testMode - Set to true to run function in test mode.
+ * @param log - A logger that the function can use for logging.
+ * @param data - An object with handler-specific contextual data.
+ * @returns True if the state was handled; false otherwise.
+ */
+export const checkChangelogs: StateHandlerFunction = async (
+	state: MachineState,
+	machine: Machine<unknown>,
+	testMode: boolean,
+	log: CommandLogger,
+	data: FluidReleaseStateHandlerData,
+): Promise<boolean> => {
+	if (testMode) return true;
+
+	const { releaseGroup, bumpType } = data;
+
+	// This check should only be run for client/server minor/major releases. Patch releases should always pass this
+	// check, as will release groups other than client/server.
+	if (["client", "server"].includes(releaseGroup) && bumpType !== "patch") {
```

Comment:
I think this code would better communicate the useful information a reader needs to understand it (and decide if its out of date) if you do something, like:

```suggestion
	// Other release groups (and independent packages) document their changes using (TODO info here) instead.
	const releaseGroupsUsingChangeLogs = ["client", "server"];
	// This check should only be run for minor/major releases.
	// Patch releases should always pass this since they (TODO: put how patches document their changes here) instead.
	if (releaseGroupsUsingChangeLogs.includes(releaseGroup) && bumpType !== "patch") {
```

This provides the assumptions this code relies on which someone can use to decide if this code is correct and/or needs to change instead of just describing that it should do what it does, which is only a small help to someone who can read the code and a burden on someone who has to change it (ex: adding another release group that uses change logs would no longer have to update it in the comment and in the code).